modules/core/site.js

```diff
@@ -1783,7 +1783,18 @@ $(function() {
         $('.reset_default_value_select').on("click", reset_default_value_select);
         $('.reset_default_value_input').on("click", reset_default_value_input);
     }
-    
+
+    $(window).on('beforeunload', function(event) {
+        if ($('#unsaved_changes').val() > 0) {
+            event.preventDefault();
+            return event.returnValue = 'Unsaved changes will be lost! Re-enter your password to save and exit';
+        }
+    });
+
+    $('a').on('click', function() {
```

Comment:
Form submissions via buttons should also stop checking for unsaved changes. In fact, clicking links from the menu is also treated as leaving the page and should warn for unsaved changes, IMHO. Only when submitting the form with changes, we should skip warning.

Reply:
@marclaporte, @kroky I just have a question on this:
I other webmail like gmail all we do is automatically saved. Many times in cypht I loose servers and profiles I created because I forget to save. Doesn't it make sense to save automatically ?

Reply:
@benoitg brought this topic up with me recently :-)

Reply:
I am fine with auto-save, or warning when-data-can-be-lost-when-leaving-a-section/page

Reply:
I agree it should be everywhere because loosing data is not cool.

Reply:
Warning on dirty forms before leaving page would be a nice addition, I agree. Autosave might be problematic in certain areas - e.g. compose form or forward form.

Reply:
https://www.npmjs.com/search?q=dirty%20form